### PR TITLE
[MRG+1] Add aarch64 wheel build support

### PR DESCRIPTION
**Pull request by odidev**
Added aarch64 wheel build support.
Related to https://github.com/alkaline-ml/pmdarima/issues/434, @aaronreidsmith Could you please review this PR?

### CODE REVIEW

**Comment by aaronreidsmith** [comment]
Thanks for opening #435 and this PR! I think we'll want to do a little more testing with this architecture before approving/merging this. Additionally, just to maximize our parallelism, GitHub Actions may not be the place for it to run.

This PR also reverts our numpy changes that were necessary to support [sktime](https://github.com/alan-turing-institute/sktime) (see #423, #425, https://github.com/alan-turing-institute/sktime/pull/828, and https://github.com/alan-turing-institute/sktime/pull/831 for context), so we will need to test with our supported version of numpy. Tagging @tgsmith61591 as well to move the conversation forward.

**Comment by odidev** [comment]
@tgsmith61591, Could you please have a look at this PR? Thanks in advance.

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 joblib>=0.11
 Cython>=0.29,!=0.29.18
-numpy~=1.19.0
+numpy>=1.19.0
```
Comment (tgsmith61591) [inline requirements.txt:3]: Same here

**Review by tgsmith61591 — APPROVED** [review]
Hey thanks so much for the PR @odidev. Sorry it took me a while to get to&mdash;I have a newborn at home and have been taking some time off. Could you please address the minor numpy versioning remark?

@aaronreidsmith when this passes, let's go ahead and merge up, then cut RCs from master just to make sure all wheels are distributed as expected.

**Comment by odidev** [comment]
Done.

File: requirements.txt (hunk 1 above)
Comment (aaronreidsmith) [reply requirements.txt:3]: @odidev looks like this file didn't get reverted in your most recent change. Once you revert, I can merge

Comment (odidev) [reply requirements.txt]: Done.

**Comment by aaronreidsmith** [comment]
Hey @odidev, sorry to give you the run around for so long on this PR. It looks like our CI keeps timing out trying to build statsmodels from scratch for aarch64. This seems to be a [known issue that they are working on](https://github.com/MacPython/statsmodels-wheels/pull/54), but until they provide pre-built wheels, I don't think we will be able to support this. Even if we did support this, users would have to build statsmodels from scratch, which is very slow. So we'll leave this PR with a `hold` label until statsmodels supports it, and we will follow that PR closely

**Comment by AGSaidi** [comment]
Looks like statsmodel has managed to release aarch64 wheels: https://pypi.org/project/statsmodels/#files 

@aaronreidsmith @odidev  perhaps we can give this another try now?

**Comment by aaronreidsmith** [comment]
Nice! @odidev if you want to rebase, we can use this branch. Or I have #458 open (which was using a conda-based approach) that we can reprupose

**Comment by AGSaidi** [comment]
stats model has, as expected, not built Python 3.6 wheels, so in addition to re-basing, we should drop Python 3.6 from the PR and add 3.10.

**Comment by odidev** [comment]
Done re-basing. Also removed python 3.6 and added python 3.10.

**Comment by aaronreidsmith** [comment]
Superseded by #458. Pretty much the same, but we changed how we test on aarch64 platforms, since that was the slow step.

**Pull request closed without merging**